app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class TweetsController < ApplicationController
+#==============================tweets================================ 
+  get '/tweets' do 
+    redirect '/login' if !logged_in?
+    
+    erb :"tweets/tweets"
+  end
+#===============================new================================== 
+  get '/tweets/new' do 
+    redirect '/login' if !logged_in?
+    
+    erb :'tweets/create_tweet'
+  end
+#-------------------------------------------------------------------- 
+  post '/tweets' do 
+    redirect '/tweets/new' if empty?(params)
+    
+    Tweet.create(content: params[:content], user_id: current_user.id)
+    redirect '/tweets'
+  end
+#==============================show================================== 
+  get '/tweets/:id' do 
+    redirect '/login' if !logged_in?
+    
+    @tweet = tweet(params[:id])
+    
+    erb :'/tweets/show_tweet'
+  end
+#==============================edit================================== 
+  get '/tweets/:id/edit' do 
+    redirect '/login' if !logged_in? 
+    
+    @tweet = tweet(params[:id])
+    
+    redirect '/tweets' if @tweet.user_id != current_user.id  
+    erb :'/tweets/edit_tweet'
+  end
+#=============================update================================= 
+  patch '/tweets/:id' do 
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...

Reply:
thanks and yea I learned about protecting these routes also after I finished this lab when doing some light reading on url hacking all my future patch/post requests will keep this in mind .. thanks :)